test/threadstest.c

```diff
@@ -464,57 +553,45 @@ static int test_multi(int idx)
  * run with a thread sanitizer, should crash if the core provider code
  * doesn't synchronize well enough.
  */
-#define MULTI_LOAD_THREADS 3
 static void test_multi_load_worker(void)
 {
     OSSL_PROVIDER *prov;
 
-    (void)TEST_ptr(prov = OSSL_PROVIDER_load(NULL, "default"));
-    (void)TEST_true(OSSL_PROVIDER_unload(prov));
+    if (!TEST_ptr(prov = OSSL_PROVIDER_load(multi_libctx, "default"))
+            || !TEST_true(OSSL_PROVIDER_unload(prov)))
+        multi_success = 0;
 }
 
 static int test_multi_load(void)
 {
-    thread_t threads[MULTI_LOAD_THREADS];
-    int i;
-
-    for (i = 0; i < MULTI_LOAD_THREADS; i++)
-        (void)TEST_true(run_thread(&threads[i], test_multi_load_worker));
-
-    for (i = 0; i < MULTI_LOAD_THREADS; i++)
-        (void)TEST_true(wait_for_thread(threads[i]));
-
-    return 1;
+    return thread_run_test(NULL, 3, &test_multi_load_worker, 0, NULL);
 }
 
 static int test_multi_default(void)
 {
-    thread_t thread1, thread2;
-    int testresult = 0;
-    OSSL_PROVIDER *prov = NULL;
-
-    multi_success = 1;
-    multi_libctx = NULL;
-    prov = OSSL_PROVIDER_load(multi_libctx, "default");
-    if (!TEST_ptr(prov))
-        goto err;
-
-    if (!TEST_true(run_thread(&thread1, thread_multi_simple_fetch))
-            || !TEST_true(run_thread(&thread2, thread_multi_simple_fetch)))
-        goto err;
-
-    thread_multi_simple_fetch();
-
-    if (!TEST_true(wait_for_thread(thread1))
-            || !TEST_true(wait_for_thread(thread2))
-            || !TEST_true(multi_success))
-        goto err;
+    return thread_run_test(&thread_multi_simple_fetch,
+                           2, &thread_multi_simple_fetch, 0, default_provider);
+}
 
-    testresult = 1;
+static void test_obj_create_one(void)
+{
+    char tids[12], oid[40], sn[30], ln[30];
+    int id = get_new_uid();
+
+    BIO_snprintf(tids, sizeof(tids), "%d", id);
+    BIO_snprintf(oid, sizeof(oid), "1.3.6.1.4.1.16604.%s", tids);
```

Comment:
For test OIDs I'd like to see us use "1.3.6.1.4.1.16604.some large constant number.test specific numbers", i.e. we introduce a large constant number (perhaps one per test file) before we have test specific numbers on the end. I'd like to reserve the possibility that at some point in the future we want to do something "real" with the 1.3.5.1.4.1.1.16604 arc without having polluted it with lots of test OIDs.

Reply:
Done.